File: Data Structures & Algorithms/binary-search/submission-0.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        // Time: O(logn)
        // Space: O(1)
        int low = 0;
        int high = nums.size() - 1;

        while(low <= high){
            int index = (low + high) / 2;

            if(nums[index] == target) return index;
            else if (nums[index] > target){
                high = index - 1;
            }
            else{
                low = index + 1;
            }
        }
        return -1;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # loop through nums and use hashmap to track numbers
        # Time: O(n) 
        # Space: O(n)
        num_indexes = {}
        for i, n in enumerate(nums):
            complement = target - n
            if complement in num_indexes:
                return [num_indexes[complement], i]
            num_indexes[n] = i


        
                